pkg/custom-provider/provider.go

```diff
@@ -147,16 +147,18 @@ func (p *prometheusProvider) buildQuery(ctx context.Context, info provider.Custo
 	// TODO: use an actual context
 	queryResults, err := p.promClient.Query(ctx, pmodel.Now(), query)
 	if err != nil {
-		klog.Errorf("unable to fetch metrics from prometheus: %v", err)
+		klog.Errorf("unable to fetch metrics from metrics source: %s", err.Error())
+		mprom.CustomMetricsFailureCounter.WithLabelValues(fmt.Sprintf("%d", err.StatusCode)).Inc()
 		// don't leak implementation details to the user
 		return nil, apierr.NewInternalError(fmt.Errorf("unable to fetch metrics"))
 	}
 
 	if queryResults.Type != pmodel.ValVector {
-		klog.Errorf("unexpected results from prometheus: expected %s, got %s on results %v", pmodel.ValVector, queryResults.Type, queryResults)
+		klog.Errorf("unexpected results from metrics source: expected %s, got %s on results %v", pmodel.ValVector, queryResults.Type, queryResults)
```

Comment:
ah I see the original logs say `prometheus`... my bad, could you revert that back